packages/flutter/test/cupertino/segmented_control_test.dart

```diff
@@ -238,63 +236,95 @@ void main() {
-        expect(textStyle.style.color, CupertinoColors.lightBackgroundGray);
-        expect(iconTheme.data.color, CupertinoColors.activeGreen);
-        expect(getBackgroundColor(tester, 0), CupertinoColors.activeGreen);
-        expect(getBackgroundColor(tester, 1), CupertinoColors.lightBackgroundGray);
+    await tester.tap(find.widgetWithIcon(IconTheme, const IconData(1)));
+    await tester.pumpAndSettle();
 
-        await tester.tap(find.widgetWithIcon(IconTheme, const IconData(1)));
-        await tester.pumpAndSettle();
+    textStyle = tester.widget(find.widgetWithText(DefaultTextStyle, 'Child 1'));
+    iconTheme = tester.widget(find.widgetWithIcon(IconTheme, const IconData(1)));
 
-        textStyle = tester.widget(find.widgetWithText(DefaultTextStyle, 'Child 1'));
-        iconTheme = tester.widget(find.widgetWithIcon(IconTheme, const IconData(1)));
+    expect(textStyle.style.color, CupertinoColors.activeGreen);
+    expect(iconTheme.data.color, CupertinoColors.lightBackgroundGray);
+    expect(getBackgroundColor(tester, 0), CupertinoColors.lightBackgroundGray);
+    expect(getBackgroundColor(tester, 1), CupertinoColors.activeGreen);
 
-        expect(textStyle.style.color, CupertinoColors.activeGreen);
-        expect(iconTheme.data.color, CupertinoColors.lightBackgroundGray);
-        expect(getBackgroundColor(tester, 0), CupertinoColors.lightBackgroundGray);
-        expect(getBackgroundColor(tester, 1), CupertinoColors.activeGreen);
+    final Offset center = tester.getCenter(find.text('Child 1'));
+    await tester.startGesture(center);
+    await tester.pumpAndSettle();
 
-        final Offset center = tester.getCenter(find.text('Child 1'));
-        await tester.startGesture(center);
-        await tester.pumpAndSettle();
+    expect(getBackgroundColor(tester, 0), const Color(0x638CFC7B));
+    expect(getBackgroundColor(tester, 1), CupertinoColors.activeGreen);
+  });
 
-        expect(getBackgroundColor(tester, 0), const Color(0x638CFC7B));
-        expect(getBackgroundColor(tester, 1), CupertinoColors.activeGreen);
-      });
+  testWidgets('Widgets are centered within segments', (WidgetTester tester) async {
+    final Map<int, Widget> children = <int, Widget>{};
+    children[0] = const Text('Child 1');
+    children[1] = const Text('Child 2');
+
+    await tester.pumpWidget(
+      new StatefulBuilder(
+        builder: (BuildContext context, StateSetter setState) {
+          return boilerplate(
+            child: new CupertinoSegmentedControl<int>(
+              key: const ValueKey<String>('Segmented Control'),
+              children: children,
+              onValueChanged: (int newValue) {},
+            ),
+          );
+        },
+      ),
+    );
+
+    final Offset topLeft = tester.getTopLeft(
+        find.byKey(const ValueKey<String>('Segmented Control'))).translate(16.0, 0.0);
+
+    expect(tester.getCenter(find.text('Child 1')),
+        getRenderSegmentedControl(tester).getChildrenAsList()[0]
```

Comment:
This is a bit indirect and needs inner knowledge of how the segmentedcontrol is built. I would black-box test it:

e.g. put the segmented control inside a 200x200 sized box. Then check that the center of child 1 is at (50, 100) and the center of child 2 is at (150, 100).